lib/discordrb/api/server.rb

```diff
@@ -200,17 +200,18 @@ def bans(token, server_id, limit = nil, before = nil, after = nil)
     )
   end
 
-  # Ban a user from a server and delete their messages from the last message_days days
+  # Ban a user from a server and delete their messages up to a given amount of time.
   # https://discord.com/developers/docs/resources/guild#create-guild-ban
-  def ban_user(token, server_id, user_id, message_days, reason = nil)
-    reason = URI.encode_www_form_component(reason) if reason
+  def ban_user(token, server_id, user_id, message_seconds, reason = nil)
     Discordrb::API.request(
       :guilds_sid_bans_uid,
       server_id,
       :put,
-      "#{Discordrb::API.api_base}/guilds/#{server_id}/bans/#{user_id}?delete_message_days=#{message_days}&reason=#{reason}",
-      nil,
-      Authorization: token
+      "#{Discordrb::API.api_base}/guilds/#{server_id}/bans/#{user_id}",
+      { delete_message_seconds: message_seconds }.to_json,
+      Authorization: token,
+      content_type: :json,
+      'X-Audit-Log-Reason': reason
     )
   end
```

Comment:
Unfortunately API scoped methods are techincally public interfaces. Let's make prime versions of these that work with seconds, and make the existing ones convert days to seconds

i.e.

```ruby
def ban_user(token, server_id, user_id, message_days, reason = nil)
    ban_user!(token, server_id, user_id, message_days * (60 * 60 * 24), reason)
end

def ban_user!(token, server_id, user_id, message_seconds, reason = nil)
   #....
 end
 ```